From 9f47dbf447162107b740eacb10fd6b1cdde3e706 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Fri, 19 Aug 2022 12:16:31 -0700
Subject: [PATCH] picocrt: Don't add _zicsr to march if already present

If the multilib architecture already includes _zicsr in the march flag,
don't add it again as that causes a compiler error.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
 picocrt/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/picocrt/meson.build b/picocrt/meson.build
index cb118b8735..800b0ddc83 100644
--- a/picocrt/meson.build
+++ b/picocrt/meson.build
@@ -49,9 +49,9 @@ foreach target : targets
   instdir = join_paths(lib_dir, value[0])
 
   if picocrt_march_add != ''
-    new_cflags=[]
+    new_cflags = []
     foreach cflag : value[1]
-      if cflag.startswith('-march')
+      if cflag.startswith('-march') and not cflag.contains(picocrt_march_add)
 	cflag = cflag + picocrt_march_add
       endif
       new_cflags += cflag
